### PR TITLE
update useLegacyNFTs

### PR DESCRIPTION
**Pull request by benisgold**
Fixes APP-####

## What changed (plus any additional context for devs)
* polygonAllowlist is now fetched as it's needed within the useLegacyNFTs queryFn. for some reason the query would still fetch when the polygon allowlist did not exist, even though its existence is a condition of the `enabled` flag. this won't have a performance impact bc of the allowlist's 10 min RQ stale time
* caches will now be deleted after 1 hour for addresses that the user does not have among their wallets (including read only), 1 week otherwise. the purpose of this is to not flood AsyncStorage, which is where all nft caches are stored.
* replaced `isFetching` with `isInitialLoading`. isFetching flickers back and forth between true and false as each page is fetched. isInitialLoading only returns true if we don't have any NFTs yet but are actively trying to fetch some
* removed `refetchInterval`, deemed unnecessary after some additional testing

### CODE REVIEW

File: src/screens/WalletScreen/index.tsx [hunk 1]
```diff
@@ -123,6 +124,8 @@ export const WalletScreen: React.FC<Props> = ({ navigation, route }) => {
     })
   );
 
+  useLegacyNFTs({ address: accountAddress });
```
Comment (estrattonbailey) [inline src/screens/WalletScreen/index.tsx:127]: Testing?

File: src/resources/nfts/index.ts [hunk 2]
```diff
@@ -86,6 +96,6 @@ export function useLegacyNFTs({ address }: { address: string }) {
   return {
     data: nfts,
     error,
-    isFetching,
+    isInitialLoading: !nfts.length && isFetching,
```
Comment (estrattonbailey) [inline src/resources/nfts/index.ts:99]: Not a blocker, but kinda odd naming imo. I think RQ might give us what we need with `isLoading`? https://tanstack.com/query/v4/docs/react/reference/useQuery

**Review by estrattonbailey — APPROVED** [review]

File: src/resources/nfts/index.ts (hunk 2 above)
Comment (estrattonbailey) [reply src/resources/nfts/index.ts:99]: Like I kinda think these hooks should _only_ return props that come from RQ, if possible. Keep them as close to RQ as possible, unless we can't.

File: src/screens/WalletScreen/index.tsx (hunk 1 above)
Comment (benisgold) [reply src/screens/WalletScreen/index.tsx:127]: oops

File: src/resources/nfts/index.ts (hunk 2 above)
Comment (benisgold) [reply src/resources/nfts/index.ts:99]: isLoading is always false, for some reason. there's no builtin prop that does what we want here. i chose this name bc useQuery actually has a prop of the same name that does the same thing: "is true whenever the first fetch for a query is in-flight". it doesn't exist on useInfiniteQuery though. I'm open to another name if you have an idea!

Comment (estrattonbailey) [reply src/resources/nfts/index.ts:99]: Nope all good!

**Pull request merged**